sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusProcessorClientIntegrationTest.java

```diff
@@ -0,0 +1,217 @@
+                message.getSequenceNumber(), message.getBody());
+            if (lastMessageReceivedTime.get() ==  null) {
+                lastMessageReceivedTime.set(OffsetTime.now());
+                countDownLatch.countDown();
+            } else {
+                long messageReceivedAfterSeconds = Duration.between(lastMessageReceivedTime.get(), OffsetTime.now()).getSeconds();
+                logger.info("Processing message again. Session: {}, Sequence #: {}. Contents: {}, message received after {} seconds.", message.getMessageId(),
+                    message.getSequenceNumber(), message.getBody(), messageReceivedAfterSeconds);
+                // Ensure that the lock is renewed and message is received again after atlest one lock renew
+                if (messageReceivedAfterSeconds >= 2 * lockTimeoutDurationSeconds) {
+                    countDownLatch.countDown();
+                }
+            }
+        } else {
+            logger.info("Received message, message id did not match. Session: %s, Sequence #: %s. Contents: %s%n", message.getMessageId(),
+                message.getSequenceNumber(), message.getBody());
+        }
+    }
+
+    private void processError(ServiceBusErrorContext context, CountDownLatch countdownLatch) {
+        logger.info("Error when receiving messages from namespace: {}. Entity: {}}",
+            context.getFullyQualifiedNamespace(), context.getEntityPath());
+    }
+
+    protected ServiceBusClientBuilder.ServiceBusProcessorClientBuilder getProcessorBuilder(boolean useCredentials,
+        MessagingEntityType entityType, int entityIndex, boolean sharedConnection) {
+
+        ServiceBusClientBuilder builder = getBuilder(useCredentials, sharedConnection);
+        switch (entityType) {
+            case QUEUE:
+                final String queueName = getQueueName(entityIndex);
+                assertNotNull(queueName, "'queueName' cannot be null.");
+
+                return builder.processor().queueName(queueName);
+            case SUBSCRIPTION:
+                final String topicName = getTopicName(entityIndex);
+                final String subscriptionName = getSubscriptionBaseName();
+                assertNotNull(topicName, "'topicName' cannot be null.");
+                assertNotNull(subscriptionName, "'subscriptionName' cannot be null.");
+
+                return builder.processor().topicName(topicName).subscriptionName(subscriptionName);
+            default:
+                throw logger.logExceptionAsError(new IllegalArgumentException("Unknown entity type: " + entityType));
+        }
+    }
+
+    protected ServiceBusClientBuilder.ServiceBusSessionProcessorClientBuilder getSessionProcessorBuilder(boolean useCredentials,
+        MessagingEntityType entityType, int entityIndex, boolean sharedConnection, AmqpRetryOptions amqpRetryOptions) {
+
+        ServiceBusClientBuilder builder = getBuilder(useCredentials, sharedConnection);
+        builder.retryOptions(amqpRetryOptions);
```

Comment:
Why do we need to override the default retry options?

Reply:
yes, for this test case. The default value is 1 minute.  For this test setting it to 30 seconds.